glommio/src/multitask.rs

```diff
@@ -128,18 +129,21 @@ impl LocalExecutor {
     pub(crate) fn spawn<T>(
         &self,
         tq: Rc<RefCell<TaskQueue>>,
-        future: impl Future<Output = T>,
+        future: impl Future<Output=T>,
     ) -> Task<T> {
         let tq = Rc::downgrade(&tq);
 
         // The function that schedules a runnable task when it gets woken up.
         let schedule = move |runnable: Runnable| {
-            let tq = tq.upgrade().unwrap();
-            {
-                let queue = tq.borrow();
-                queue.ex.local_queue.push(runnable);
+            let tq = tq.upgrade();
+
+            if let Some(tq) = tq {
+                {
+                    let queue = tq.borrow();
+                    queue.ex.local_queue.push(runnable);
+                }
```

Comment:
Fantastic work!

I am wondering if this one is really necessary, though ?
If I understood this correctly, we no longer remove the timer if the reactor is not present from the timer side. So why do we have to test for queues being still in existence here ?

Reply:
Otherwise, we would panic, on the test above because the task queue is absent. We schedule task on waker drop, but we can not because the queue is already dropped.

Reply:
But in the case of the "background" task, the only thing which is left during the drop of the executor is waker. If such a task is paused of course.

Reply:
ok, that makes sense.